packages/@sanity/components/src/utilities/tryFindScrollContainer.js

```diff
@@ -0,0 +1,31 @@
+/* global window, document */
+export default function tryFindScrollContainer(element, callback) {
+  if (!window || !document) {
+    return false
+  }
+
+  if (!element) {
+    console.error('tryFindScrollContainer: No element', element) //eslint-disable-line
+  }
+
+  let scrollContainer = element
+  let foundScrollContainer = false
+
+  while (!foundScrollContainer) {
+    if (!scrollContainer || !scrollContainer.parentNode) {
+      break
+    }
+
+    if (['overlay', 'auto', 'scroll'].includes(window.getComputedStyle(scrollContainer).overflowY)) {
+      foundScrollContainer = true
+      break
+    }
+    scrollContainer = scrollContainer.parentNode
+  }
+
+  if (scrollContainer === document) {
+    scrollContainer = document.body
+  }
+
+  return callback(scrollContainer)
```

Comment:
I think its better if you just return the scrollContainer here, and where you use it, you can pass it directly to the callback, like this:
```js
callback(tryFindScrollContainer(element))
```
```js
// e.g instead of.
tryFindScrollContainer(element, this.setScrollContainerElement)
// you can just do
this.setScrollContainerElement(tryFindScrollContainer(element))
```

Reply:
👍